lock_api/src/lib.rs

```diff
@@ -93,6 +93,8 @@
 #![warn(rust_2018_idioms)]
 #![cfg_attr(feature = "nightly", feature(const_fn))]
 
+// REVIEW: would recommend unconditionally tagging this crate as `#![no_std]`
```

Comment:
That results in this warning when building `std`:
```
warning: unused attribute===========================================>  ] 31/32: std                                                                                              
  --> src/libstd/../parking_lot/lock_api/src/lib.rs:91:1
   |
91 | #![no_std]
   | ^^^^^^^^^^
   |
   = note: #[warn(unused_attributes)] on by default
```

Which is why I introduced the condition.

Reply:
I think that's indicative of a different organization being needed in that case? For example `src/lib.rs` could be something like:

```
mod lock_api;
pub use lock_api::*;
```

and then `src/lock_api/mod.rs` is the entry point for the crate and libstd